pkg/tests/tasks/security/authorization/jwt_test.go

```diff
@@ -18,157 +18,196 @@ import (
-		msg, err := util.PodExec("foo", sleepPod, "sleep", cmd, true)
-		util.Inspect(err, "Failed to get response", "", t)
-		if !strings.Contains(msg, "401") {
-			log.Log.Errorf("Verify denied request Unexpected response: %s", msg)
-			t.Errorf("Verify denied request Unexpected response: %s", msg)
-		} else {
-			log.Log.Infof("Success. Get expected response: %s", msg)
-		}
-
-		log.Log.Info("Verify a request without a JWT is allowed")
-		cmd = fmt.Sprintf(`curl "http://httpbin.%s:8000/headers" -sS -o /dev/null -w "%%{http_code}\n"`, "foo")
-		msg, err = util.PodExec("foo", sleepPod, "sleep", cmd, true)
-		util.Inspect(err, "Failed to get response", "", t)
-		if !strings.Contains(msg, "200") {
-			log.Log.Errorf("Verify request without JWT Unexpected response: %s", msg)
-			t.Errorf("Verify request without JWT Unexpected response: %s", msg)
-		} else {
-			log.Log.Infof("Success. Get expected response: %s", msg)
-		}
-	})
+		t.Log("This test validates authorization policies with JWT Token")
 
-	t.Run("Security_authorization_allow_JWT_requestPrincipal", func(t *testing.T) {
-		defer util.RecoverPanic(t)
+		t.LogStep("Install httpbin and sleep")
+		app.InstallAndWaitReady(t, app.Httpbin(ns), app.Sleep(ns))
 
-		log.Log.Info("Apply a policy requires all requests to have a valid JWT")
-		util.KubeApplyContents("foo", JWTRequireRule)
-		time.Sleep(time.Duration(50) * time.Second)
+		t.LogStep("Check if httpbin returns 200 OK when no authorization policies are in place")
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			oc.Exec(t,
+				pod.MatchingSelector("app=sleep", ns),
+				"sleep",
+				httpbinRequest("GET", "/ip"),
+				assert.OutputContains(
+					"200",
+					"Got expected 200 OK from httpbin",
+					"Expected 200 OK from httpbin, but got a different HTTP code"))
+		})
 
-		log.Log.Info("Download JWT token")
 		jwtURL := "https://raw.githubusercontent.com/istio/istio/release-1.9/security/tools/jwt/samples/demo.jwt"
-		token, err := util.ShellMuteOutput(`curl %s -s`, jwtURL)
+		token, err := util.Shell(`curl %s -s`, jwtURL)
 		token = strings.Trim(token, "\n")
-		// token, err = util.Shell(`echo %s | cut -d '.' -f2 - | base64 --decode -`, token)
-		util.Inspect(err, "Failed to get JWT token", "", t)
-
-		log.Log.Info("Verify request with a valid JWT")
-		cmd := fmt.Sprintf(`curl "http://httpbin.%s:8000/headers" -sS -o /dev/null -H "Authorization: Bearer %s" -w "%%{http_code}\n"`, "foo", token)
-		msg, err := util.PodExec("foo", sleepPod, "sleep", cmd, true)
-		util.Inspect(err, "Failed to get response", "", t)
-		if !strings.Contains(msg, "200") {
-			log.Log.Errorf("Verify request with valid JWT Unexpected response: %s", msg)
-			t.Errorf("Verify request with valid JWT Unexpected response: %s", msg)
-		} else {
-			log.Log.Infof("Success. Get expected response: %s", msg)
+		if err != nil {
+			t.Error("message")
 		}
 
```

Comment:
You should call `token := curl.Request(t, jwtURL, nil)` here. But first, you need to refactor that function to return the response body instead of `time.Duration`. It doesn't look like that duration is used anywhere, so you can change the function without having to change any of the tests.

Reply:
FYI: I already refactored the function so it now returns the body. When you rebase, you should be able to use it.